src/containers/VideoFeed.jsx

```diff
@@ -0,0 +1,94 @@
+import React, { Component } from 'react';
+import { API, Auth, Storage } from 'aws-amplify';
+import { Container, Spinner } from 'react-bootstrap';
+import VideoItem from './VideoItem';
+
+class VideoFeed extends Component {
+  constructor(props) {
+    super(props);
+    this.state = { videos: null };
+  };
+
+  static defaultProps = {
+     numVideos: "5",
+     owner: null, // if not sent as prop: loads videos from all users
+   };
+
+  async componentDidMount(props) {
+    this.debugger();
+
+    try {
+      const videos = await API.get('videocloud', '/videos');
+      this.filterQuery(videos);
+
+      console.log("- all videos in DB: ", videos)
+    } catch (error) {
+      console.log(error);
+    }
+  }
+
+  async debugger() {
+    try {
+      const user = await Auth.currentAuthenticatedUser();
+      const list = await Storage.list('', { level: 'private' });
+
+      console.log("- all accessable S3 videos for user", user.attributes['custom:id']+":", list);
+    } catch (error) {
+      console.log(error);
+    }
+  }
+
+  compDate(a, b) {
+    return new Date(b.createdAt).getTime() - new Date(a.createdAt).getTime();
+  }
+
+  filterQuery(videos, props) {
+    let filtered = videos;
+    // if profile page:   show only videos from the user of the profile
+    // else if home page: show videos from all users
+    if (this.props.owner != null) {
+      filtered = filtered.filter(row =>
+        row.owner === parseInt(this.props.owner)
+      );
+    }
+
+    // Filters the newest videos
```

Comment:
This comment suggests that you can extract the lines below into a function: `filterVideos()`.